modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/validation/ParameterModelValidator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.introspection.validation;
+
+import static org.mule.extension.api.introspection.ParameterModel.RESERVED_NAMES;
+import static org.mule.util.Preconditions.checkArgument;
+
+import org.mule.extension.api.exception.IllegalModelDefinitionException;
+import org.mule.extension.api.introspection.ConfigurationModel;
+import org.mule.extension.api.introspection.ConnectionProviderModel;
+import org.mule.extension.api.introspection.ExtensionModel;
+import org.mule.extension.api.introspection.OperationModel;
+import org.mule.extension.api.introspection.ParameterModel;
+
+public class ParameterModelValidator implements ModelValidator
```

Comment:
add javadoc by the example of the existing validators (don't forget the @since.4.0)